pkg/sentry/server/validator/internal/common.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+
+	sentryv1pb "github.com/dapr/dapr/pkg/proto/sentry/v1"
+)
+
+// Validate validates the common rules for all requests.
+func Validate(_ context.Context, req *sentryv1pb.SignCertificateRequest) (spiffeid.TrustDomain, error) {
+	var errs []error
+
+	// TODO: we should also validate that the app ID is alpha numeric, and
+	// doesn't contain things like spaces etc.
+	if len(req.Id) == 0 {
+		errs = append(errs, errors.New("app ID is required"))
```

Comment:
If you use errors.Join, all of these can be changed to:

```go
var err error
if len(req.Id) == 0 {
	err = errors.Join(err, errors.New("app ID is required"))
}
```

It's ok to call `errors.Join` and "join" a nil error too, which will not be added